src/index.js

```diff
@@ -328,5 +346,87 @@ module.exports.create = function (opts) {
 };
 
 module.exports.version = function (cb) {
-    run(_command('version'), cb);
+    module.exports._run(_command('version'), cb);
+};
+
+module.exports.boxAdd = function (box, args, cb) {
+    if (typeof box !== 'string' && cb) {
+        cb('box must be provided as a string');
+    }
+    cb = cb || args;
+
+    var command = _command(['box', 'add', '-f'], args, box);
+    var proc = module.exports._run(command, cb);
+
+    var emitter = new EventEmitter;
+    proc.stdout.on('data', function (buff) {
+        var data = buff.toString();
+
+        var res = parsers.downloadStatusParser(data);
+        if (res) {
+            emitter.emit('progress', res.machine, res.progress, res.rate, res.remaining);
+        }
+    });
+    return emitter;
+};
+
+module.exports.boxList = function (args, cb) {
+    cb = cb || args;
+
+    var command = _command(['box', 'list'], args);
+    module.exports._run(command, function (err, out) {
+        if (err) {
+            return cb(err);
+        }
+        cb(null, parsers.boxListParser(out));
+    });
+};
+
+module.exports.boxOutdated = function (args, cb) {
+    cb = cb || args;
+
+    var command = _command(['box', 'outdated', '--global'], args);
+    module.exports._run(command, cb);
+};
+
+module.exports.boxPrune = function (args, cb) {
+    cb = cb || args;
+
+    var command = _command(['box', 'prune', '-f'], args);
+    module.exports._run(command, cb);
+};
+
+module.exports.boxRemove = function (name, args, cb) {
+    if (typeof name !== 'string' && cb) {
+        cb('name must be provided as a string');
+    }
+
+    cb = cb || args;
+
+    var command = _command(['box', 'remove', '-f'], args, name);
+    module.exports._run(command, cb);
+};
+
+module.exports.boxUpdate = function (box, provider, cb) {
+    if (typeof box !== 'string' && cb) {
+        cb('box must be provided as a string');
+    }
+
+    if (typeof provider !== 'string' && cb) {
+        cb('provider must be provided as a string');
```

Comment:
412 and 416 same `return cb(`

Reply:
Or in case of this function, we might want to return the `EventEmitter` just if case the client expects that type. But we don't want to call the `cb` twice.

Reply:
Ya, this case is different. I returned blank event emitters. Should keep a client happy in this scenario.